apis/nucleus/src/utils/generateSetExpression.js

```diff
@@ -0,0 +1,234 @@
+const FilterType = {
```

Comment:
This file should have tests, I see that there are some for the original one, so bring them over.